Formula/elm-format.rb

```diff
@@ -15,27 +15,23 @@ class ElmFormat < Formula
   end
 
   depends_on "cabal-install" => :build
-  depends_on "ghc@8.8" => :build
+  depends_on "haskell-stack" => :build
+  depends_on arch: :x86_64 # no ghc (OSX,AArch64) binary via `haskell-stack`
 
-  def build_elm_format_conf
-    <<~EOS
-      module Build_elm_format where
-
-      gitDescribe :: String
-      gitDescribe = "#{version}"
-    EOS
-  end
+  uses_from_macos "xz" => :build # for `haskell-stack` to extract ghc
 
   def install
-    defaults = buildpath/"generated/Build_elm_format.hs"
-    defaults.write(build_elm_format_conf)
-
-    (buildpath/"elm-format").install Dir["*"]
-
-    cd "elm-format" do
-      system "cabal", "v2-update"
-      system "cabal", "v2-install", *std_cabal_v2_args
-    end
+    # Currently, dependency constraints require an older `ghc` patch version than available
+    # in Homebrew. Try using Homebrew `ghc` on update. Optionally, consider adding `ghcup`
+    # as a lighter-weight alternative to `haskell-stack` for installing particular ghc version.
+    ENV.deparallelize { system "stack", "setup", "-j#{ENV.make_jobs}", "--stack-root", buildpath/".stack" }
+    ENV.prepend_path "PATH", Dir[buildpath/".stack/programs/*/ghc-*/bin"].first
+    system "cabal", "v2-update"
+
+    # Directly running `cabal v2-install` fails: Invalid file name in tar archive: "avh4-lib-0.0.0.1/../"
+    # Instead, we can use the upstream's build.sh script, which utilizes the Shake build system.
+    system "./build.sh", "--", "build"
```

Comment:
Same failure seen in previous attempt #70750. We might be missing something by running `cabal v2-install` while bypassing Shake. Similar to manually creating `generated/Build_elm_format.hs`, which would have been generated by Shake.

Upstream shell script is https://github.com/avh4/elm-format/blob/0.8.5/build.sh
Expanded version of commands run:
```sh
PKG_ENV_FILE="_build/shake-package-env-$(ghc --numeric-version)"
mkdir -p _build
cabal v2-install --package-env "$PKG_ENV_FILE" --lib shake
ghc --make Shakefile.hs -package-env "$PKG_ENV_FILE" \
    -XLambdaCase \
    -rtsopts -threaded -with-rtsopts=-I0 \
    -outputdir=_build -o _build/build
_build/build -- build
```

Not sure of all flags used and which commands are run parallel.